proto/eth/v2/custom.go

```diff
@@ -0,0 +1,62 @@
+package eth
+
+import (
+	types "github.com/prysmaticlabs/prysm/v3/consensus-types/primitives"
+	ethpbv1 "github.com/prysmaticlabs/prysm/v3/proto/eth/v1"
+	"math/bits"
+)
+
+const (
+	NextSyncCommitteeIndex = uint64(55)
+	FinalizedRootIndex     = uint64(105)
+)
+
+type Update interface {
+	GetAttestedHeader() *ethpbv1.BeaconBlockHeader
+	GetNextSyncCommittee() *SyncCommittee
+	GetNextSyncCommitteeBranch() [][]byte
+	GetFinalizedHeader() *ethpbv1.BeaconBlockHeader
+	SetFinalizedHeader(*ethpbv1.BeaconBlockHeader)
+	GetFinalityBranch() [][]byte
+	GetSyncAggregate() *ethpbv1.SyncAggregate
+	GetSignatureSlot() types.Slot
+}
+
+// TODO: move this somewhere common
+func FloorLog2(x uint64) int {
+	return bits.Len64(uint64(x - 1))
+}
+
+var _ Update = (*FinalityUpdate)(nil)
```

Comment:
Not sure if this is needed.

Reply:
This ensures FinalityUpdate is implementing the interface, there's no other way to enforce that at compile time afaik

Reply:
I see.  Thanks.